### ISSUE TITLE
Integration with substitutes

### ISSUE THREAD

**Issue by MrTomek** [rico-et22/elektronik-timetable#21]
Please add an alert or an exclamation mark icon next to a lesson on the timetables of classes, teachers and classrooms.

Data loaded from:

https://elektronik.rzeszow.pl/uploads/zastepstwa/InformacjeOZastepstwach.pdf  
or  
https://elektronik.rzeszow.pl/api/replacements.json (ready scraper PDF used for TV in the lobby)

**Comment by MrTomek** [comment]
Example view:

![image](https://github.com/rico-et22/elektronik-timetable/assets/13391813/ada8d39b-5ddb-4055-af2a-87865c381d6b)

**Comment by rico-et22** [comment]
In my opinion, it would be better to replace the original lesson with a substitute lesson plus an exclamation mark (something like school e-journals do), rather than just placing a mark on the original lesson that wouldn't provide direct information about the substitute.

> https://elektronik.rzeszow.pl/api/replacements.json (ready scraper PDF used for TV in the lobby)
We already use it for the substitutions table 😉

**Comment by olekbaran** [comment]
I think we can also add a tooltip while hovering on an exclamation mark with some info about original lesson

**Comment by felpcereti** [comment]
Hello I was doing something similar to that thing you have planned but there's one problem. It's my second time working with a react application so the code looks bad.
![obraz](https://github.com/rico-et22/elektronik-timetable/assets/39708080/c17dd12d-0c41-454c-a48a-f27bc654e1cc)

My fork: https://github.com/rico-et22/elektronik-timetable/compare/main...felpcereti:elektronik-timetable:main

**Comment by rico-et22** [comment]
@felpcereti 
That way of implementing this is what I actually was thinking of!

I consulted the "exclamation mark" way with other contributors and it would be too clunky and require additional clicks to display the replacement data.

So, we prefer it your way, as we were planning to.

However, I tried launching your fork and it gets me this error:
![image](https://github.com/rico-et22/elektronik-timetable/assets/28688352/9f801fe8-e38b-4366-826d-e21ad4009b83)
with the following env variables:
```
NEXT_PUBLIC_TIMETABLE_BASE_URL=https://zse.rzeszow.pl/plan-lekcji
NEXT_PUBLIC_REPLACEMENTS_API_URL=https://www.elektronik.rzeszow.pl/api/replacements.json
```

Please fix the code so that it can be used with ZSE Rzeszów's official data sources first. Then try making a pull request

**Comment by felpcereti** [comment]
I'm using the same env variables.
I changed following lines of code https://github.com/felpcereti/elektronik-timetable/commit/56daa8de8215279c2b2ada6adbd91b4d97ac9f2b#diff-e253519adf5ed18c5b9535b50c8f3553796ba50482c17df141c00ac7d3935074L103
```ts
day.length -
          1 -
          day
            .slice()
            .reverse()
            .findIndex((dayHour) => dayHour.length > 0)
```
to 
```ts
day.findLastIndex((dayHour) => dayHour.length > 0)
```

It's a newer method which was added in node version 18.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLastIndex#browser_compatibility

If that's a problem I can just revert the change.

**Comment by felpcereti** [comment]
Also I wonder if you have any other means of communication like discord, telegram or session.

**Comment by rico-et22** [comment]
@felpcereti 
My discord: rico_et22#5855

Oh, I was still using Node 16, upgraded to 18 and the error is gone.

I think you can make a pull request (prepare for a code review), but before, there are two **urgent** things to fix:
1. **top priority** - fix the responsiveness & font size of the table's links as on smaller sizes (eg. 1024px) the rendering breaks.
Remove text-sm from here:
![image](https://github.com/rico-et22/elektronik-timetable/assets/28688352/339996ce-c07e-4349-b15d-223a15892a21)
and add w-1/2 class to this container, when displaying a teacher's name:
![image](https://github.com/rico-et22/elektronik-timetable/assets/28688352/88cc2e20-a865-4de2-a732-6167fe821354)

2.Increase spacing between groups (both table and list)

**Comment by rico-et22** [comment]
Done. PR #24 and #25 .
@felpcereti thanks!

**Comment by MrTomek** [comment]
In my opinion, you still need to add this on the teacher's timetable `/teacher`.

**Comment by rico-et22** [comment]
@MrTomek 
> In my opinion, you still need to add this on the teacher's timetable `/teacher`.

Teacher pages fixed in #27
